### PR TITLE
Remove version/help dependency on Environment

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
Current tiup will fail to print `tiup --version` nor `tiup xxx --help` while local data is not correct.

### What is changed and how it works?
Move env init to `PersistentPreRunE`. We can run `tiup --version`, `tiup --help`, `tiup xxx --help` even when there is no network nor `root.json`. However, other commands like `tiup install`, help commands like `tiup help mirror` still can't be used.

Example(without local root.json):
image
image
image

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -122,17 +124,25 @@ the latest stable version will be downloaded from the repository.`,
...
 		cmd, n, e := cmd.Root().Find(args)
-		if (cmd == rootCmd || e != nil) && len(n) > 0 {
+		if (cmd == rootCmd || e != nil) && len(n) > 0 && err != nil {
 			externalHelp(env, n[0], n[1:]...)
 		} else {
 			cmd.InitDefaultHelpFlag() // make possible 'help' flag to be shown
 			_ = cmd.Help()
 		}
+		if err != nil {
+			fmt.Printf("\ninit local env meet error: %s\n", err)
```
Comment (lichunzhu) [inline cmd/root.go:136]: BTW, I want to delete this line. Users can get this error when they try to execute some commands.

Comment (lonng) [reply cmd/root.go:136]: I think it's ok to remove it.

File: pkg/environment/tiup.go [hunk 2]
```diff
@@ -11,20 +11,16 @@
...
-	tiupmeta "github.com/pingcap/tiup/pkg/environment"
-)
+var _env *Environment
 
-var _env *tiupmeta.Environment
-
-// SetTiupEnv the gloable env used.
-func SetTiupEnv(env *tiupmeta.Environment) {
+// SetTiupEnv the global env used.
+func SetTiupEnv(env *Environment) {
```
Comment (lonng) [inline pkg/environment/tiup.go:19]: How about changing this to `SetGlobalEnv`?

File: pkg/environment/tiup.go [hunk 3]
```diff
@@ -11,20 +11,16 @@
...
 	_env = env
 }
 
-// TiupEnv Get the gloable env used.
-func TiupEnv() *tiupmeta.Environment {
+// TiupEnv Get the global env used.
+func TiupEnv() *Environment {
```
Comment (lonng) [inline pkg/environment/tiup.go:24]: ditto, `GlobalEnv()`.

File: cmd/root.go [hunk 4]
```diff
@@ -122,17 +124,23 @@ the latest stable version will be downloaded from the repository.`,
...
+		env, err := environment.InitEnv(repoOpts)
 		cmd, n, e := cmd.Root().Find(args)
-		if (cmd == rootCmd || e != nil) && len(n) > 0 {
+		if (cmd == rootCmd || e != nil) && len(n) > 0 && err != nil {
 			externalHelp(env, n[0], n[1:]...)
 		} else {
 			cmd.InitDefaultHelpFlag() // make possible 'help' flag to be shown
 			_ = cmd.Help()
 		}
+		if err == nil {
```
Comment (lonng) [inline cmd/root.go:135]: Do we need to set the global env? I think there will not be used in the subsequent.

**Pull request merged**